### ISSUE TITLE
First floating device has (many) more mooring lines than other devices in array

### ISSUE THREAD

**Issue by KD379** [DTOcean/dtocean#39]
Hello and Happy New Year!

This is not an issue or bug, I am just looking for advice or an explanation for some results I was not expecting. Apologies in advice for the long-winded explanation.
I have been using DTOcean to see how changes to ULS safety factors may affect mooring and foundation system design with regards to reliability and economics.

**Step by step**

- New project (Saved as PROJECT 27)
- Device technology type: wave floating
- Select database: DTOcean_examples
- Initiate pipeline
- Select site: Eureka CA
- Select device: RM3
- Initiate bathymetry
- Add modules: Hydrodynamics, Mooring and Foundations
- Add assessments: Economics, Reliability
- Initiate dataflow
- Filled in/changed some simulation inputs:
   o	Device rated power: 0.286 MW
   o	Array rated power: 2.86 MW (modelling array of 10 devices)
   o	Preferred foundation type: none
   o	Mooring system type: none
   o	Mooring ALS safety factor: 1.0 (constant)
   o	Mooring ULS safety factor: 1.0-2.25 in 0.25 increments (variable)

-  Ran 6 identical simulations, changing only the ULS safety factor.  I.e.: 

   1.	27A: ULS safety factor = 1.0
   2.	27B: ULS safety factor = 1.25
   3.	27C: ULS safety factor = 1.5
   4.	27D: ULS safety factor = 1.75
   5.	27E: ULS safety factor = 2.0
   6.	27F: ULS safety factor = 2.25

- Compiled following results from each simulation on Excel:
 _Economics_
         	CAPEX cost breakdown (to see mooring and foundations (M&F)   CAPEX)
_Reliability_
         	Mooring and Foundations sub-system reliability metrics
_Mooring and Foundations_
         	Moorings and Foundations bill of materials
         	Foundations components supplementary data
         	Mooring line supplementary data

**What I Expected**
I expected that as the ULS safety factor was increased, both 1) the mooring and foundation CAPEX and 2) the mean time to first M&F failure would also increase. In other words, I predicted that as I increased the ULS safety factor, the M&F system would be more reliable and in turn, the M&F expenditure would increase to accommodate the greater safety factor.

**Actual Results**
However, in reality, the MTTF and CAPEX do not increase in correlation with the ULS. And the higher CAPEX tends to give a lower reliability estimate and vice versa. The following graph displays these results. I have also included an attachment should anyone wish to see a more in-depth look at the results.
![ResultsGraphKD](https://user-images.githubusercontent.com/72255727/104751303-203e8a80-574d-11eb-8723-b23d7001a455.png)
[PROJECT 27.xlsx](https://github.com/DTOcean/dtocean/files/5821667/PROJECT.27.xlsx)

**Query**
In conclusion, I was hoping someone (more knowledgable about mooring & foundation design than me!) may be able to shed some light on these results and let me know if the results are accurate or if they think there is a problem with my data, inputs, etc. Whether or not the results are correct is my main concern as I hope to use them for a project.

Thanks very much in advance for the help!
-Kate

**Comment by H0R5E** [comment]
Hi @KD379, happy new year to you, also. Yeah, I would say your overall results are surprising. Would you be able to share the DTO files for the 1 and 2.25 cases? From some of the array layout diagrams it looks like the moorings module has not been able to find a solution, but has still completed (there seems to be two devices without any moorings at all for the 2.25 case), so the costings are just wrong.

It's always been a bit weird in these cases, I'm afraid. It seems to throw the kitchen sink at one device and then not deal with the other ones properly.

**Comment by KD379** [comment]
Hi @H0R5E. I have shared the PROJECT27.dto file on this drive (I think you already have access).

My mistake regarding the 2.25 case where it looks like some devices have no foundations, I forgot to select the excess data for the diagram when the number of foundations increased when I was copying and pasting the pages - rookie mistake! So I uploaded the excel file on the drive too with the diagram fixed.

You'll also notice in the .dto file there is a 27G in which I tried to run the simulation with ULS safety factor = 2.5 but DTOcean couldn't find a solution and I got an error 'A Runtime Error: Maximum number of lines (12) exceeded'.

https://drive.google.com/drive/folders/1UYzsiH8Br7ePcmkbZ5FGMeVUvS978sgN

**Comment by H0R5E** [comment]
@KD379, thanks for the data. I'm sorry, but I'm pretty time limited this week, but I will try to have a look at it before next week.

**Comment by H0R5E** [comment]
@KD379, I've attached some files that should help with the CAPEX part of your problem. You need to replace the files in the `DTOcean\Lib\site-packages\dtocean_moorings` directory with the ones in the following zip:

[moorings.zip](https://github.com/DTOcean/dtocean/files/5867119/moorings.zip)

I'm afraid this might not still provide everything you want. Essentially the mooring design algorithm is a bit ropey (pun intended) and it struggles to purposely design the components. So, for the first device, if it fails to find a set of components that will work for the design number of lines, it adds a new mooring line and tries again. The thing is, on a lot of occasions, extra lines are not needed, just a different combination of components, but it can only find these using additional lines (because the dynamics change slightly). That's why, when it uses the same components found using multiple lines for the other devices, not as many lines are needed.

So, what I've done is told the algorithm to redesign the first device again after it finds the initial solution, so that it will be similar to the other devices in the farm. In theory this should give you some relationship between ULS safety factor and CAPEX, but it will be weak <s>as I think the costs are dominated more by the foundations than by the mooring lines</s> as all the designs seem to be using the most expensive chain, regardless of ULS. Also, it's not guaranteed as the program might just overdesign for a particular ULS because it struggles to find the optimal set. 

Unfortunately, even with this fix, you won't see meaningful results for trend in the MTTF. That's because DTOcean doesn't calculate the impact of the operating environment on the reliability of the components, nor is there any modification based on safety factor. This is important, and I think it's something DTOceanPlus is looking at, but note that their mooring designs are much simpler also, so it's a bit easier to do. It would be interesting to see what you might expect the relationship of safety factor to MTTF to be. Here is a [search](http://letmegooglethat.com/?q=factor+of+safety+and+reliability) that has some articles available on the subject.

EDIT: The strange MTTF results you were getting was because of the (random number of ) extra lines in the first device of each array. The overall MTTF is very sensitive to the number of points of failure in the system. Also, the CAPEX was being dominated by these extra lines, but because it is kind of random how many are added, the results were not consistent.

Anyway, I think this is all useful information to report, as these problems are not very well documented for DTOcean as it stands.

Hope that helps a little,

Mat

**Comment by KD379** [comment]
Hi Mat,

First of all, thanks for looking into this in so much depth! 

I will replace those files and rerun the simulations. It will be interesting to see how the moorings and foundations for the first device turns out now! I could post the CAPEX vs. ULS safety factor on this thread after if you wanted to have a look.

I see now how I was barking up the wrong tree trying to find a meaningful relationship between the safety factors and the reliability assessment. I was taking reliability to mean structural reliability analysis (i.e. system is no longer reliable when the load is greater than the structural strength) and thus thought applying the greater safety factor would increase the system strength and in turn its reliability, whereas now I understand DTOcean is analysing system reliability statistically using component failure rates. Reliability is a word that seems to be thrown around in literature quite often under different definitions which added to the confusion but the DTOcean Deliverable 4.6 framework paper cleared this up nicely!

Thanks a million,
Kate

**Comment by H0R5E** [comment]
Fixed by https://github.com/DTOcean/dtocean-moorings/commit/2c9e353f991666a33d142adb7b03af904b56522f
